### PR TITLE
8255947: [macos] Signed macOS jpackage app doesn't filter spurious '-psn' argument

### PR DESCRIPTION
**Pull request by sashamatveev**
This is regression from JDK-8242302 and for some reason removing -psn argument code was removed during refactoring. Fixed be adding removing -psn argument back. Also, test was added to test this functionality.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux x64 | Linux x86 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- | ----- |
| Build | ⏳ (3/5 running) | ⏳ (2/2 running) | ⏳ (2/2 running) | ⏳ (2/2 running) |

### Issue
 * [JDK-8255947](https://bugs.openjdk.java.net/browse/JDK-8255947): [macos] Signed macOS jpackage app doesn't filter spurious '-psn' argument

### Reviewers
 * [Andy Herrick](https://openjdk.java.net/census#herrick) (@andyherrick - **Reviewer**) ⚠️ Review applies to 7df8d2251c63c7fff66e759812321ab9fc4a95d2
 * [Alexey Semenyuk](https://openjdk.java.net/census#asemenyuk) (@alexeysemenyukoracle - Committer)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1154/head:pull/1154`
`$ git checkout pull/1154`

### CODE REVIEW

File: src/jdk.jpackage/macosx/native/common/MacSysInfo.cpp [hunk 1]
```diff
@@ -77,4 +80,15 @@ tstring getProcessModulePath() {
     return FileUtils::toAbsolutePath(reply);
 }
 
+tstring_array getCommandArgs(CommandArgProgramNameMode progNameMode) {
+    tstring_array result;
+    for (int i = progNameMode == ExcludeProgramName ? 1 : 0; i < argc; i++) {
+        if (strncmp(argv[i], PSN_ARG, strlen(PSN_ARG)) == 0) {
```
Comment (alexeysemenyukoracle) [inline src/jdk.jpackage/macosx/native/common/MacSysInfo.cpp:86]: I'd suggest to use more C++:
`const tstring psnArgPrefix = "-psn_";
for (int i = progNameMode == ExcludeProgramName ? 1 : 0; i < argc; i++) {
    const tstring arg = argv[i];
    if (!tstrings::startsWith(arg, psnArgPrefix)) {
        result.push_back(arg);
    }
}`

File: src/jdk.jpackage/macosx/native/common/MacSysInfo.cpp [hunk 2]
```diff
@@ -77,4 +80,15 @@ tstring getProcessModulePath() {
...
+            continue;
+        }
+        result.push_back(argv[i]);
+    }
```
Comment (alexeysemenyukoracle) [inline src/jdk.jpackage/macosx/native/common/MacSysInfo.cpp:90]: I'd suggest to use C++ here:
```suggestion
    const tstring psnArgPrefix = "-psn_";
    for (int i = progNameMode == ExcludeProgramName ? 1 : 0; i < argc; i++) {
        const tstring arg = argv[i];
        if (!tstrings::startsWith(arg, psnArgPrefix)) {
            result.push_back(arg);
        }
    }
```

**Comment by grischard** [comment]
For bug reproduction, note that the very latest version of macOS (Big Sur, 11.0.1), released just yesterday, doesn't pass the `-psn_x_yyyy` argument anymore

**Review by andyherrick — APPROVED** [review]
Looks good

**Review by alexeysemenyukoracle — APPROVED** [review]

**Pull request closed without merging**